flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/planner/delegation/hive/HiveParserDMLHelper.java

```diff
@@ -154,11 +154,12 @@ public HiveParserDMLHelper(
         RelDataTypeFactory typeFactory = plannerContext.getTypeFactory();
         LinkedHashMap<String, RelDataType> targetColToCalcType = new LinkedHashMap<>();
         List<TypeInfo> targetHiveTypes = new ArrayList<>();
-        TableSchema tableSchema =
-                HiveParserUtils.fromUnresolvedSchema(destTable.getUnresolvedSchema());
-        String[] fieldNames = tableSchema.getFieldNames();
+        ResolvedSchema resolvedSchema =
+                catalogManager.resolveCatalogTable(destTable).getResolvedSchema();
```

Comment:
After removed, this test can not pass(because we get an unresolved Schema when retrieve from hive), I'm not sure how you access to this `During decoupling Hive connector, we won't pass the ability to call method catalogManager.resolveCatalogTable to HiveParser.` But, currently, the catalogManager is widely used in the HiveParser. Do you think it's feasible to improve this part when doing the decoupling ? 

<img width="1238" alt="image" src="https://user-images.githubusercontent.com/9486140/225352134-c459079a-4cb7-4f8a-85a4-2dbf5db5d669.png">

Reply:
I want to introduce [CatalogRegistry](https://github.com/apache/flink/pull/22166/files#diff-46be322b211295390bc999d0312835c5eac5442aef284faa473e75cb2fc78ddcR27) which will contain limited and necessary interface related to CatalogManger. Since it's a public interface,  I really don't want to expose much.

Back to this case, as far as i can see,  it only happens for `create  table as` statement, you can reolve the `schema` to `ResolvedSchema` in [here](https://github.com/apache/flink/blob/981ef8bd874ac5c28512d8abfa17adcefa44e9bb/flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/planner/delegation/hive/parse/HiveParserDDLSemanticAnalyzer.java#L1072).

Remember we can just convert `UnresolvedPhysicalColumn` to `PhysicalColumn` and throw excpetion directly for other kind of `UnresolvedColumn`. It'll be simple to do the conversion.

Reply:
Or we can just create a resolved version table from the `org.apache.flink.table.catalog.hive.util.HiveTableUtil#createSchema()` ? Since they are all physical column and the dataType have been converted.